src/backend/api/controllers/exports.ts

```diff
@@ -27,6 +27,42 @@ const exportsController = {
     const userId = await expectUserId(req);
     const scheduleId = stringToObjectId(req.params.scheduleId, "scheduleId");
 
+    const startDateYear: number | undefined = maybeStringToNumber(
```

Comment:
if you use `stringToNumber` instead here, you don't need to do the check for each thing being defined below
also, since the return type of the function is explicit, you can leave out the `number | undefined` type def here